File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/op/atts/value/impl/AttIdValue.java
package ru.citeck.ecos.records3.record.op.atts.value.impl;

import ru.citeck.ecos.commons.utils.MandatoryParam;
import ru.citeck.ecos.records3.record.op.atts.value.AttValue;

public class AttIdValue implements AttValue {

    private final String id;

    public AttIdValue(Object id) {
        MandatoryParam.check("id", id);
        this.id = id.toString();
    }

    @Override
    public String getString() {
        return id;
    }

    @Override
    public String getId() {
        return id;
    }
}

File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/atts/value/impl/AttIdValue.kt
package ru.citeck.ecos.records3.record.atts.value.impl

import ru.citeck.ecos.records3.record.atts.value.AttValue

class AttIdValue(private val id: Any?) : AttValue {

    override fun asText(): String? {
        return id?.toString() ?: ""
    }

    override fun getId(): Any {
        return id ?: ""
    }
}
